test/jasmine/tests/polar_test.js

```diff
@@ -496,3 +499,153 @@ describe('Test relayout on polar subplots:', function() {
+            }
+        });
+    }
+
+    function assertEventCnt(expected, msg) {
+        eventNames.forEach(function(k) {
+            var m = msg + ' event cnt for ' + k;
+
+            if(k in expected) {
+                expect(eventCnts[k]).toBe(expected[k], m);
+            } else {
+                expect(eventCnts[k]).toBe(0, m);
+            }
+        });
+    }
+
+    function _hover(pos) {
+        eventData = '';
+        mouseEvent('mousemove', pos[0], pos[1]);
+    }
+
+    function _unhover(pos) {
+        eventData = '';
+        mouseEvent('mouseout', pos[0], pos[1]);
+    }
+
+    function _click(pos) {
+        eventData = '';
+        gd._mouseDownTime = 0;
+        click(pos[0], pos[1]);
+    }
+
+    function _doubleClick(pos) {
+        gd._mouseDownTime = 0;
+        eventData = '';
+        return doubleClick(pos[0], pos[1]);
+    }
+
+    it('should trigger hover/unhover/click/doubleclick events', function(done) {
+        var fig = Lib.extendDeep({}, require('@mocks/polar_scatter.json'));
+        var ptPos = [250, 200];
+        var blankPos = [200, 120];
+        var marginPos = [20, 20];
+
+        function _assert(ptExpectation, cntExpecation, msg) {
+            if(Array.isArray(ptExpectation)) {
+                assertEventPointData(ptExpectation, msg);
+            } else {
+                expect(eventData).toBe(ptExpectation, msg);
+            }
+            assertEventCnt(cntExpecation, msg);
+        }
+
+        _plot(fig)
+        .then(function() { _hover(ptPos); })
+        .then(function() {
+            _assert([{
+                r: 3.26,
+                theta: 68.08
+            }], {
+                plotly_hover: 1
+            }, 'after hover on pt');
+        })
+        .then(function() { _unhover(blankPos);})
+        .then(function() {
+            _assert([{
+                r: 3.26,
+                theta: 68.08
+            }], {
+                plotly_hover: 1,
+                plotly_unhover: 1
+            }, 'after unhover off pt');
+        })
+        .then(function() { _hover(marginPos);})
+        .then(function() {
+            _assert('', {
+                plotly_hover: 1,
+                plotly_unhover: 1,
+            }, 'after hovering in margin');
+        })
+        .then(function() { _click(ptPos); })
+        .then(function() {
+            _assert([{
+                r: 3.26,
+                theta: 68.08
+            }], {
+                plotly_hover: 2,
+                plotly_unhover: 1,
+                plotly_click: 1
+            }, 'after click');
+        })
+        .then(function() { return _doubleClick(ptPos); })
+        .then(function() {
+            assertEventCnt({
+                plotly_hover: 2,
+                plotly_unhover: 1,
+                plotly_click: 3,
+                plotly_doubleclick: 1,
+                plotly_relayout: 1
+            }, 'after doubleclick');
+        })
+        .catch(fail)
+        .then(done);
```

Comment:
Great test, very readable!

Presumably polar also handles right-clicks the same way as cartesian? Want to 🔒 it ala https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/click_test.js#L173-L213?

Reply:
done in https://github.com/plotly/plotly.js/pull/2200/commits/473643cf8dfb12f1543bd8dcb01ca00c568f6398